src/util/customTransformer-v1.js

```diff
@@ -54,63 +54,48 @@ async function userTransformHandlerV1(
-    const tags = {
-      transformerVersionId: userTransformation.versionId,
-      identifier: 'v1',
-      ...metaTags,
-    };
-
-    logger.debug(`Isolate VM being created... `);
-    const isolatevmFactory = await getFactory(
-      userTransformation.code,
-      libraryVersionIds,
-      userTransformation.versionId,
-      userTransformation.secrets || {},
-      testMode,
-    );
-    const isolatevm = await isolatevmFactory.create();
-    logger.debug(`Isolate VM created... `);
+  if (!userTransformation.versionId) {
+    return { transformedEvents : events };
+  }
 
-    // Transform the event...
-    stats.counter('events_to_process', events.length, tags);
-    const isolateStartWallTime = calculateMsFromIvmTime(isolatevm.isolateStartWallTime);
-    const isolateStartCPUTime = calculateMsFromIvmTime(isolatevm.isolateStartCPUTime);
+  const isolatevmFactory = await getFactory(
+    userTransformation.code,
+    libraryVersionIds,
+    userTransformation.versionId,
+    userTransformation.secrets || {},
+    testMode,
+  );
 
-    const invokeTime = new Date();
-    let transformedEvents;
-    // Destroy isolatevm in case of execution errors
-    try {
-      transformedEvents = await transform(isolatevm, events);
-    } catch (err) {
-      logger.error(`Error encountered while executing transformation: ${err.message}`);
-      isolatevmFactory.destroy(isolatevm);
-      throw err;
-    }
-    const { logs } = isolatevm;
-    stats.timing('run_time', invokeTime, tags);
-    const isolateEndWallTime = calculateMsFromIvmTime(isolatevm.isolate.wallTime);
-    const isolateEndCPUTime = calculateMsFromIvmTime(isolatevm.isolate.cpuTime);
+  logger.debug(`Creating IsolateVM`);
+  const isolatevm = await isolatevmFactory.create();
 
-    //TODO: fix "Value is not a valid number: NaN" error and uncomment
-    //stats.timing('isolate_wall_time', isolateEndWallTime - isolateStartWallTime, tags);
-    //stats.timing('isolate_cpu_time', isolateEndCPUTime - isolateStartCPUTime, tags);
+  const invokeTime = new Date();
+  let transformedEvents;
+  let logs;
+  let transformationError;
 
-    // Destroy the isolated vm resources created
-    logger.debug(`Isolate VM being destroyed... `);
+  try {
+    transformedEvents = await transform(isolatevm, events);
+  } catch (err) {
+    logger.error(`Error encountered while executing transformation: ${err.message}`);
+    transformationError = err;
+    throw err;
+  } finally {
+    logger.debug(`Destroying IsolateVM`);
+    logs = isolatevm.logs;
     isolatevmFactory.destroy(isolatevm);
-    logger.debug(`Isolate VM destroyed... `);
-
-    return { transformedEvents, logs };
-    // Events contain message and destination. We take the message part of event and run transformation on it.
-    // And put back the destination after transforrmation
+    // send the observability stats
+    const tags = {
+      identifier: 'v1',
+      errored: transformationError ? true : false,
```

Comment:
```suggestion
      errored: transformationError ? 'true' : 'false',
```
stat values are string or numbers

Reply:
https://github.com/rudderlabs/rudder-transformer/blob/0947c58948a8192ac16f1a7e9b5b7520d5dc7530/src/util/customTransformer-faas.js#L90

I followed the testMode stat and it is boolean. I'll check if the testMode ( boolean ) implicitly gets converted to the string